kingpin/actors/aws/ecs.py

```diff
@@ -366,87 +422,528 @@ def _tasks_done(self, tasks):
+
+    If the service already exists, it is upgraded.
 
-def _load_task_definition(task_definition_file, parameters):
-    """Loads and verifies a task definition template file, and interpolates
-    parameters.
+    **Options**
 
-    Args:
-        task_definition_file: task definition file to load.
-        parameters: dict of key/value pairs to interpolate into the file.
+    :region:
+      AWS region (or zone) name, such as us-west-2 or eu-west-1.
+
+    :cluster:
+      ECS cluster in the region to launch the task into.
+
+    :task_definition:
+      String of path to the Task Definition file template.
+      Must be a local file path.
+      Tokens to be interpolated must be of the form %VAR%.
+
+    :service_definition:
+      String of path to the Service Definition file template.
+      Must be a local file path.
+      Tokens to be interpolated must be of the form %VAR%.
+      Implicit fields - do not include these:
+      'serviceName', 'taskDefinition', 'desiredCount', 'clientToken'
+      Allowed fields: 'loadBalancers', 'role', 'deploymentConfiguration'
+      Default: None.
+
+    :service_name:
+      Service name to use. If not specified, this will use
+      the Task Definition's family.
+      Default: None.
+
+    :tokens:
+      A dictionary of key/value pairs used to fill in the tokens for the
+      Task and Service Definition template. Default: {}.
+
+    :count:
+      How many instances of the service to deploy.
+      Default: 1.
+
+    :wait:
+      Whether to wait for the services to deploy.
+      Default: True.
+
+    **Examples**
+
+    .. code-block:: yaml
+
+       actor: aws.ecs.Service
+       desc: Deploy taskworker
+       options:
+          task_definition: nextdoor-tw.yaml
+          region: us-west-2
+          cluster: us1-internal
+
+    **Dry Mode**
 
-    Returns:
-        Parsed and interpolated task definition dict.
+    Will only validate and interpolate tokens into both the
+    task and service definition files.
     """
-    task_definition_file = utils.convert_script_to_dict(
-        task_definition_file, parameters)
-    try:
-        jsonschema.validate(task_definition_file,
-                            TASK_DEFINITION_SCHEMA)
-    except jsonschema.exceptions.ValidationError as e:
-        raise exceptions.InvalidOptions(e)
-    return task_definition_file
+
+    all_options = {
+        'region': (str, REQUIRED,
+                   'AWS region (or zone) name, like us-west-2 or eu-west-1.'),
+        'cluster': (str, REQUIRED,
+                    'ECS cluster in the region to launch the task into.'),
+        'task_definition':
+            (str, REQUIRED,
+             'String of path to the Task Definition file template. '
+             'Must be a local file path. '
+             'Tokens to be interpolated must be of the form %VAR%.'),
+        'service_definition':
+            (str, None,
+             'String of path to the Service Definition file template.'
```

Comment:
Since all these quoted strings are concatenated together, add spaces at he end of them.